src/peer.rs

```diff
@@ -78,14 +64,14 @@ pub(crate) mod test_utils {
     // Generate Vec<Peer> where no two peers have the same name.
     pub(crate) fn arbitrary_unique_peers(
         count: impl Into<SizeRange>,
-        age: impl Strategy<Value = u8>,
     ) -> impl Strategy<Value = Vec<Peer>> {
-        proptest::collection::btree_map(arbitrary_xor_name(), (any::<SocketAddr>(), age), count)
-            .prop_map(|peers| {
+        proptest::collection::btree_map(arbitrary_xor_name(), any::<SocketAddr>(), count).prop_map(
```

Comment:
Maybe `arbitrary_xor_name` should take the age strategy as a parameter? Because right now it generates names with completely random ages, including invalid ages like 0, 1, 2 and 3. That might break some tests.